cylc/flow/job_runner_mgr.py

```diff
@@ -568,8 +568,8 @@ def _job_submit_impl(
         # Create NN symbolic link, if necessary
         self._create_nn(job_file_path)
         for name in JOB_LOG_ERR, JOB_LOG_OUT:
-            with suppress(OSError):
-                os.unlink(os.path.join(job_file_path, name))
+            with suppress(FileNotFoundError):
+                os.unlink(os.path.join(os.path.dirname(job_file_path), name))
```

Comment:
Perhaps pull the dirname out of the loop for a minor speedup? Or would python be smart enough to do that itself? You could even do a tweak to _create_nn to accept the dirname rather than the job path so it doesn't get determined in multiple spots, `_create_nn` only needs the `01` type path.